x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/bulk/BulkShardOperationsTests.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ccr.action.bulk;
+
+import org.apache.lucene.index.Term;
+import org.elasticsearch.action.support.replication.TransportWriteAction;
+import org.elasticsearch.cluster.routing.RecoverySource;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.cluster.routing.ShardRoutingState;
+import org.elasticsearch.cluster.routing.TestShardRouting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.VersionType;
+import org.elasticsearch.index.mapper.Uid;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.index.shard.IndexShardTestCase;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.index.translog.Translog;
+import org.elasticsearch.xpack.ccr.CcrSettings;
+import org.elasticsearch.xpack.ccr.index.engine.FollowingEngineFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class BulkShardOperationsTests extends IndexShardTestCase {
+
+    private static final byte[] SOURCE = "{}".getBytes(StandardCharsets.UTF_8);
+
+    // test that we use the primary term on the follower when applying operations from the leader
+    public void testPrimaryTermFromFollower() throws IOException {
+        final Settings settings = Settings.builder().put(CcrSettings.CCR_FOLLOWING_INDEX_SETTING.getKey(), true).build();
+        final IndexShard followerPrimary = newStartedShard(true, settings, new FollowingEngineFactory());
+
+        // we use this primary on the operations yet we expect the applied operations to have the primary term of the follower
+        final long primaryTerm = randomLongBetween(1, followerPrimary.getPrimaryTerm());
```

Comment:
nit: we can do this fully random - there's no relationship between the primary term of the leader and the follower

Reply:
The randomness was not meant to imply there was such a relationship. 😇 

I pushed 0a8499bf1b2d7d749f7ea9b855c6efac3d5fa9ef.